app/consumer.py

```diff
@@ -26,57 +26,67 @@
 from app.helpers.exceptions import BadMessageError, DecryptError, RetryableError
 
 
-def get_delivery_count_from_properties(properties):
-    """
-    Returns the delivery count for a message from the rabbit queue. The
-    value is auto-set by rabbitmq.
-    """
-    delivery_count = 0
-    if properties.headers and 'x-delivery-count' in properties.headers:
-        delivery_count = properties.headers['x-delivery-count']
-    return delivery_count + 1
-
-
 class Consumer(AsyncConsumer):
 
     def __init__(self, args=None, cfg=None):
         self._args = args
         self._cfg = cfg
-        self.quarantine_publisher = QueuePublisher(logger, settings.RABBIT_URLS, settings.RABBIT_QUARANTINE_QUEUE)
+        self.quarantine_publisher = QueuePublisher(logger,
+                                                   settings.RABBIT_URLS,
+                                                   settings.RABBIT_QUARANTINE_QUEUE)
         super().__init__()
 
+    def get_delivery_count_from_properties(self, properties):
+        """
+        Returns the delivery count for a message from the rabbit queue. The
+        value is auto-set by rabbitmq.
+        """
+        delivery_count = 0
+        if properties.headers and 'x-delivery-count' in properties.headers:
+            delivery_count = properties.headers['x-delivery-count']
+        return delivery_count + 1
+
+    def get_tx_id_from_properties(self, properties):
+        """
+        Returns the tx_id for a message from a rabbit queue. The value is
+        auto-set by rabbitmq.
+        """
+        return properties.headers.get('tx_id')
+
     def on_message(self, unused_channel, basic_deliver, properties, body):
 
-        delivery_count = get_delivery_count_from_properties(properties)
+        delivery_count = self.get_delivery_count_from_properties(properties)
+        tx_id = self.get_tx_id_from_properties(properties)
```

Comment:
Should we be logging/handling something if the `tx_id` is missing from the header? It should always be there (by the time this goes live anyhow) so it _not_ being there should be an indication that something is ary or the message on the queue isn't what it seems to be.

Maybe log and push the message to quarantine?